PocketKit/Sources/PocketKit/MyList/SortFilterMenu/View/SortMenuHeaderView.swift

```diff
@@ -0,0 +1,93 @@
+import UIKit
+
+class SortMenuHeaderView: UITableViewHeaderFooterView {
+    static let identifier = "SortMenuHeaderView"
+    private lazy var sortIconImageView: UIImageView = {
+        let sortIconIV = UIImageView(image: UIImage(asset: .sort))
+        sortIconIV.translatesAutoresizingMaskIntoConstraints = false
+        sortIconIV.contentMode = .scaleAspectFit
+        sortIconIV.widthAnchor.constraint(equalToConstant: SortMenuHeaderView.Constants.SortIcon.width).isActive = true
+        sortIconIV.heightAnchor.constraint(equalToConstant: SortMenuHeaderView.Constants.SortIcon.height).isActive = true
+        return sortIconIV
+    }()
+    
+    private lazy var sortTitleLabel: UILabel = {
+        let sortTitleLbl = UILabel()
+        sortTitleLbl.textColor = UIColor(.ui.grey5)
+        return sortTitleLbl
+    }()
+    
+    override init(reuseIdentifier: String?) {
+        super.init(reuseIdentifier: reuseIdentifier)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension SortMenuHeaderView {
+    
+    func configureHeaderView(headerTitle: String) {
+        let containerView = UIView()
+        containerView.heightAnchor.constraint(equalToConstant: SortMenuHeaderView.Constants.ContainerView.height).isActive = true
+        addSubview(containerView)
+        setLayoutConstraints(containerView: self, subView: containerView)
+
+        sortTitleLabel.attributedText = NSAttributedString(string: headerTitle, style: .header.sansSerif.h8)
+
+        let stackViewContainer = UIStackView()
+        stackViewContainer.axis = .horizontal
+        stackViewContainer.spacing = SortMenuHeaderView.Constants.StackViewContainer.spacing
+        stackViewContainer.alignment = .fill
+        stackViewContainer.distribution = .fillProportionally
+        stackViewContainer.addArrangedSubview(sortIconImageView)
+        stackViewContainer.addArrangedSubview(sortTitleLabel)
+        containerView.addSubview(stackViewContainer)
+        let stackViewPadding = SortMenuPadding(left: SortMenuHeaderView.Constants.StackViewContainer.leftPadding)
+        setLayoutConstraints(containerView: containerView, subView: stackViewContainer, padding: stackViewPadding)
```

Comment:
Preferably, the view hierarchy would already be setup during the initializer of this class. Then the responsibility of this method would simply be to update the content.
If we have this method adding subviews and rearranging views, then we would end up adding more subviews every time the cell is re-used.

Reply:
Makes sense, thank you.